ompi/mpi/fortran/mpif-h/ialltoallw_f.c

```diff
@@ -116,13 +116,10 @@ void ompi_ialltoallw_f(char *sendbuf, MPI_Fint *sendcounts,
                             c_recvtypes, c_comm, &c_request);
     if (NULL != ierr) *ierr = OMPI_INT_2_FINT(c_ierr);
     if (MPI_SUCCESS == c_ierr) *request = PMPI_Request_c2f(c_request);
+    c_request->req_flags |= OMPI_REQ_NB_RELEASE_DATA;
```

Comment:
don't we have a race condition here?
if yes, I am afraid we already have the same kind of race condition in `ompi/mpi/c/ialltoallw.c`

Reply:
The race condition exists only if the request is completed in the initialization function and the completion callback is triggered from inside (instead of being triggered during the request release in MPI_Test* or MPI_Wait*). I'm afraid that in order to avoid the race condition we need to change the API of the nonblocking collectives to add a flag, such that the collective itself does the handling of the temporary arrays. I was hoping to avoid this approach, as it is invasive and require a change in the collective module version.